restapi/api/reserve_endpoints_test.go

```diff
@@ -122,9 +124,147 @@ func TestValidateDeviceNotReserved(t *testing.T) {
 	validateNotReserved(t, responseRecorder)
 }
 
+func TestValidateMiddlewareHeaderMissing(t *testing.T) {
+	r = setupRouter()
+	responseRecorder := httptest.NewRecorder()
+
+	launchAppRequest, err := http.NewRequest("POST", "/"+randomDeviceUDID+"/launch?bundleID=com.apple.Preferences", nil)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	r.ServeHTTP(responseRecorder, launchAppRequest)
+	require.Equal(t, http.StatusBadRequest, responseRecorder.Code, "Code should be BadRequest if X-GO-IOS-RESERVE header is missing")
+
+	var response GenericResponse
+	responseData, _ := ioutil.ReadAll(responseRecorder.Body)
+	err = json.Unmarshal(responseData, &response)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	require.NotEmpty(t, response.Error, "There is no error message returned when X-GO-IOS-RESERVE header is missing")
+}
+
+func TestValidateMiddlewareHeaderEmpty(t *testing.T) {
+	r = setupRouter()
+	responseRecorder := httptest.NewRecorder()
+
+	launchAppRequest, err := http.NewRequest("POST", "/"+randomDeviceUDID+"/launch?bundleID=com.apple.Preferences", nil)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	launchAppRequest.Header.Add("X-GO-IOS-RESERVE", "")
+	r.ServeHTTP(responseRecorder, launchAppRequest)
+	require.Equal(t, http.StatusBadRequest, responseRecorder.Code, "Code should be BadRequest if X-GO-IOS-RESERVE header is empty")
+
+	var response GenericResponse
+	responseData, _ := ioutil.ReadAll(responseRecorder.Body)
+	err = json.Unmarshal(responseData, &response)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	require.NotEmpty(t, response.Error, "There is no error message returned when X-GO-IOS-RESERVE header is missing")
```

Comment:
Could be good also to verify the correct HTTP Status codes are sent.

Reply:
that is actually validated a few lines above, in all tests I check response code first before proceeding with other checks :P

Reply:
🤦  never mind me